src/aspire/basis/polar_2d.py

```diff
@@ -81,22 +82,19 @@ def evaluate(self, v):
         v, sz_roll = unroll_dim(v, 2)
         nimgs = v.shape[1]
 
-        half_size = self.nrad * self.ntheta // 2
+        half_size = self.ntheta // 2
 
         v = m_reshape(v, (self.nrad, self.ntheta, nimgs))
 
-        v = (v[:, :self.ntheta // 2, :]
-             + v[:, self.ntheta // 2:, :].conj())
+        v = (v[:, :half_size, :]
+             + v[:, half_size:, :].conj())
 
-        v = m_reshape(v, (half_size, nimgs))
+        v = m_reshape(v, (self.nrad*half_size, nimgs))
+        x = np.empty((self.sz[0], self.sz[1], nimgs), dtype='complex128', order='F')
+
+        for isample in range(0, nimgs):
```

Comment:
Do we have an issue reminding us about this (and other places) where a many call is needed?

Reply:
Added a TODO in the text.

Reply:
I will create an issue and assign to myself when I get internet. Thanks